source/base/parameter_handler.cc

```diff
@@ -2311,13 +2319,391 @@ ParameterHandler::print_parameters (std::ostream     &out,
+  recursively_print_parameters (std::vector<std::string>(), // start at the top level
+                                style,
+                                0,
+                                out);
 
   return out;
 }
 
 
 
+void
+ParameterHandler::recursively_print_parameters (const std::vector<std::string> &target_subsection_path,
+                                                const OutputStyle               style,
+                                                const unsigned int              indent_level,
+                                                std::ostream                   &out) const
+{
+  AssertThrow (out, ExcIO());
+
+  Assert ((style != XML) && (style != JSON),
+          ExcMessage ("This function should not be necessary for XML or JSON output."));
+
+  const boost::property_tree::ptree &current_section
+    = entries->get_child (collate_path_string(target_subsection_path));
+
+  unsigned int overall_indent_level = indent_level;
+
+  switch (style)
+    {
+    case Text:
+    case ShortText:
+    {
+      // first find out the longest entry name to be able to align the
+      // equal signs to do this loop over all nodes of the current
+      // tree, select the parameter nodes (and discard sub-tree nodes)
+      // and take the maximum of their lengths
+      std::size_t longest_name = 0;
+      for (boost::property_tree::ptree::const_iterator
+           p = current_section.begin();
+           p != current_section.end(); ++p)
+        if (is_parameter_node (p->second) == true)
+          longest_name = std::max (longest_name,
+                                   demangle(p->first).length());
+
+      // likewise find the longest actual value string to make sure we
+      // can align the default and documentation strings
+      std::size_t longest_value = 0;
+      for (boost::property_tree::ptree::const_iterator
+           p = current_section.begin();
+           p != current_section.end(); ++p)
+        if (is_parameter_node (p->second) == true)
+          longest_value = std::max (longest_value,
+                                    p->second.get<std::string>("value").length());
+
+
+      // print entries one by one. make sure they are sorted by using
+      // the appropriate iterators
+      bool first_entry = true;
+      for (boost::property_tree::ptree::const_assoc_iterator
+           p = current_section.ordered_begin();
+           p != current_section.not_found(); ++p)
+        if (is_parameter_node (p->second) == true)
+          {
+            const std::string value = p->second.get<std::string>("value");
+
+            // if there is documentation, then add an empty line
+            // (unless this is the first entry in a subsection), print
+            // the documentation, and then the actual entry; break the
+            // documentation into readable chunks such that the whole
+            // thing is at most 78 characters wide
+            if ((!(style & 128)) &&
```

Comment:
I don't think we have documented this. In fact, I'd very much like to remove this obscure feature. Would you mind letting me doing this as a follow-up?